src/shared/tpm2-util.c

```diff
@@ -2354,14 +2354,20 @@ int tpm2_pcr_mask_from_string(const char *arg, uint32_t *ret_mask) {
                 if (r < 0)
                         return log_error_errno(r, "Failed to parse PCR list: %s", arg);
 
-                r = safe_atou(pcr, &n);
-                if (r < 0)
-                        return log_error_errno(r, "Failed to parse PCR number: %s", pcr);
-                if (n >= TPM2_PCRS_MAX)
-                        return log_error_errno(SYNTHETIC_ERRNO(ERANGE),
+                if (pcr && ascii_isalpha(pcr[0])) {
```

Comment:
please drop this, just use DEFINE_STRING_TABLE_LOOKUP_FROM_STRING_FALLBACK() if you use that instead of plain DEFINE_PRIVATE_STRING_TABLE_LOOKUP_FROM_STRING() then the numeric parsing will be done for you automatically

Reply:
> please drop this, just use DEFINE_STRING_TABLE_LOOKUP_FROM_STRING_FALLBACK() if you use that instead of plain DEFINE_PRIVATE_STRING_TABLE_LOOKUP_FROM_STRING() then the numeric parsing will be done for you automatically

Alright sir.
Will study it and implement

Reply:
> please drop this, just use DEFINE_STRING_TABLE_LOOKUP_FROM_STRING_FALLBACK() if you use that instead of plain DEFINE_PRIVATE_STRING_TABLE_LOOKUP_FROM_STRING() then the numeric parsing will be done for you automatically

@poettring I think I have implemented all the changes you requested apart from this.
Kindly confirm sir.

Reply:
```C
DEFINE_STRING_TABLE_LOOKUP_WITH_FALLBACK(pcr_index, int, TPM2_PCRS_MAX);
```